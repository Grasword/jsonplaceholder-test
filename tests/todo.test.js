import ToDoModel from '../model/todoModel'

describe('Todo', () => {
  test('remote API returns 200', async () => {
    const { status } = await ToDoModel.fetch()
    expect(status).toBe(200)
  })

  test('return data is an object', async () => {
    const { data } = await ToDoModel.fetch()
    expect(data).toMatchSnapshot()
  })

  test('formatted todo complies to requirements', async () => {
    const response = await ToDoModel.fetch()
    const todo = ToDoModel.formatTodo(response.data)
    expect(todo).toMatchSnapshot()
  })

  test('completed should be boolean and title should be string', async () => {
    const response = await ToDoModel.fetch()
    const todo = ToDoModel.formatTodo(response.data)
    todo.forEach((t) => {
      expect(typeof t.title).toBe('string')
      expect(typeof t.completed).toBe('boolean')
    })
  })

  test('returns 120 items', async () => {
    const { data } = await ToDoModel.fetch(120)
    expect(data.length).toBe(120)
  })
})
