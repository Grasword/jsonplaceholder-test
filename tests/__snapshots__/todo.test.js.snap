// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo return data is an object 1`] = `
Object {
  "completed": false,
  "id": 1,
  "title": "delectus aut autem",
  "userId": 1,
}
`;
